Opsworks Provider: added custom json options

dieend opened the pull request:
Added custom_json options. If not set, will fallback to default configuration.
User have responsibility to set the custom_json value by himself. I assumed anyone who wanted to use custom_json already understand how to uses it.

closes #271

jasny commented:
Please add the `custom_json` option to README.md.

Also you probably shouldn't edit `lib/dpl/version.rb`.

P.S. To run your own fork of use `edge: local`. See http://stackoverflow.com/questions/30584467/how-can-i-run-a-custom-version-of-dpl-on-travis-ci

dieend commented:
whoops. accidentally added the version.rb in the pull request. I changed the version in reference to #146. 

fixed it. Also added the `custom_json` in readme.

BanzaiMan commented on lib/dpl/provider/ops_works.rb:
```diff
@@ -32,16 +32,20 @@ def check_auth
-              migrate: !!options[:migrate],
-              scm: {
-                revision: current_sha
+        if options[:custom_json]
```
This `if` seems rather wordy.

How about `options[:custom_json] || { }`.

dieend commented:
@BanzaiMan updated.
